sqlglot/optimizer/canonicalize.py

```diff
@@ -26,7 +26,7 @@ def canonicalize(expression: exp.Expression) -> exp.Expression:
 
 def add_text_to_concat(node: exp.Expression) -> exp.Expression:
     if isinstance(node, exp.Add) and node.type and node.type.this in exp.DataType.TEXT_TYPES:
-        node = exp.Concat(this=node.this, expression=node.expression)
+        node = exp.Concat(expressions=[node.left, node.right])
```

Comment:
This looks like a bug, `CONCAT` didn't have `this` and `expression` as args, I wonder how it used to work.

Reply:
I guess we didn't error out before because `concat_sql` relies on `function_fallback_sql` under the hood and so it'd pick up args not defined in `arg_types`:

```python
    def function_fallback_sql(self, expression: exp.Func) -> str:
        args = []
        for arg_value in expression.args.values():
            if isinstance(arg_value, list):
                for value in arg_value:
                    args.append(value)
            else:
                args.append(arg_value)

        return self.func(expression.sql_name(), *args)
```